clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientBuilderImpl.java

```diff
@@ -294,8 +307,10 @@ public ZeebeClientBuilder overrideAuthority(final String authority) {
 
   @Override
   public ZeebeClient build() {
-    applyOverrides();
-    applyDefaults();
+    if (applyEnvironmentVariableOverrides) {
+      applyOverrides();
+      applyDefaults();
```

Comment:
:x: It seems `applyDefaults` is actually just setting up the OAuth credentials provider if one of the environment variables is set. Can we just inline `applyDefaults` into `applyOverrides`? I don't want someone adding things later in `applyDefaults` thinking this is always going to be executed, and have a flag which disable environment overrides somehow turn that off.

Reply:
Makes total sense to me! I also invested some time to figure out if applyDefaults needs to run or not :-) Added